kaldb/src/test/java/com/slack/kaldb/logstore/search/LogIndexSearcherImplTest.java

```diff
@@ -382,26 +382,157 @@ public void testFullTextSearch() {
     msg2.addProperty("field2", "1234");
     strictLogStore.logStore.addMessage(msg2);
+    strictLogStore.logStore.commit();
+    strictLogStore.logStore.refresh();
+    // Search using _all field.
+    assertThat(
+            strictLogStore
+                .logSearcher
+                .search(TEST_DATASET_NAME, "_all:baby", 0, MAX_TIME, 1000, 1)
+                .hits
+                .size())
+        .isEqualTo(1);
+    assertThat(
+            strictLogStore
+                .logSearcher
+                .search(TEST_DATASET_NAME, "_all:1234", 0, MAX_TIME, 1000, 1)
+                .hits
+                .size())
+        .isEqualTo(2);
+    // Default all field search.
+    assertThat(
+            strictLogStore
+                .logSearcher
+                .search(TEST_DATASET_NAME, "baby", 0, MAX_TIME, 1000, 1)
+                .hits
+                .size())
+        .isEqualTo(1);
+    assertThat(
+            strictLogStore
+                .logSearcher
+                .search(TEST_DATASET_NAME, "1234", 0, MAX_TIME, 1000, 1)
+                .hits
+                .size())
+        .isEqualTo(2);
 
     final LogMessage msg3 =
         makeMessageWithIndexAndTimestamp(
             3, "baby car 1234", TEST_DATASET_NAME, time.plusSeconds(4));
     strictLogStore.logStore.addMessage(msg3);
     strictLogStore.logStore.commit();
     strictLogStore.logStore.refresh();
+    // Search using _all field.
+    assertThat(
+            strictLogStore
+                .logSearcher
+                .search(TEST_DATASET_NAME, "_all:baby", 0, MAX_TIME, 1000, 1)
+                .hits
+                .size())
+        .isEqualTo(2);
+    assertThat(
+            strictLogStore
+                .logSearcher
+                .search(TEST_DATASET_NAME, "_all:1234", 0, MAX_TIME, 1000, 1)
+                .hits
+                .size())
+        .isEqualTo(3);
+    // Default all field search.
+    assertThat(
+            strictLogStore
+                .logSearcher
+                .search(TEST_DATASET_NAME, "baby", 0, MAX_TIME, 1000, 1)
+                .hits
+                .size())
+        .isEqualTo(2);
+    assertThat(
+            strictLogStore
+                .logSearcher
+                .search(TEST_DATASET_NAME, "1234", 0, MAX_TIME, 1000, 1)
+                .hits
+                .size())
+        .isEqualTo(3);
+
+    // empty string
+    assertThat(
+            strictLogStore
+                .logSearcher
+                .search(TEST_DATASET_NAME, "", 0, MAX_TIME, 1000, 1)
+                .hits
+                .size())
+        .isEqualTo(3);
 
-    SearchResult<LogMessage> babyInAll =
-        strictLogStore.logSearcher.search(TEST_DATASET_NAME, "_all:baby", 0, MAX_TIME, 1000, 1);
-    assertThat(babyInAll.hits.size()).isEqualTo(2);
+    // Currently, returns empty since we don't parse wild card queries.
+    // TODO: One we start parsing wild card queries this should return 3.
```

Comment:
This isn't accurate - for fields that are analyzed wildcard queries do work today. You can see this if you query for "app*"  in this test you should have results for "apple". Additionally if this is just "*" it returns the entire dataset when the field is analyzed.

Reply:
Good catch. Updated the tests for wildcard queries. * throws an exception. If we want * to not throw an exception, we should update the query parser to convert that to empty query.